src/PostgREST/Query.hs

```diff
@@ -84,6 +83,16 @@ data QueryResult
   | MaybeDbResult InspectPlan  (Maybe (TablesMap, RoutineMap, Maybe Text))
   | NoDbResult    InfoPlan
 
+-- The Queries that run on every request
+data MainQuery = MainQuery
+  { mqTxVars  :: SQL.Snippet       -- ^ the transaction variables that always run on each query
+  , mqPreReq  :: Maybe SQL.Snippet -- ^ the pre-request function that runs if enabled
+  , mqCount   :: SQL.Snippet       -- ^ this count query is actually a fragment of the main query, but the same count query also runs after the main one in the case of `count=estimated`, so it's cached here.
+  -- TODO only one of the following queries actually runs on each request, once OpenAPI is removed from core it will be easier to refactor this
+  , mqMain    :: SQL.Snippet
+  , mqOpenAPI :: (SQL.Snippet, SQL.Snippet, SQL.Snippet)
+  }
```

Comment:
This type is the main change here, it's now produced at the App.hs level and then passed to the main transaction.